assert/assertions.go

```diff
@@ -512,22 +534,44 @@ func samePointers(first, second interface{}) bool {
 // to a type conversion in the Go grammar.
 func formatUnequalValues(expected, actual interface{}) (e string, a string) {
 	if reflect.TypeOf(expected) != reflect.TypeOf(actual) {
-		return fmt.Sprintf("%T(%s)", expected, truncatingFormat(expected)),
-			fmt.Sprintf("%T(%s)", actual, truncatingFormat(actual))
+		return fmt.Sprintf("%T(%s)", expected, truncatingFormat(expected, false)),
+			fmt.Sprintf("%T(%s)", actual, truncatingFormat(actual, false))
 	}
 	switch expected.(type) {
 	case time.Duration:
 		return fmt.Sprintf("%v", expected), fmt.Sprintf("%v", actual)
 	}
-	return truncatingFormat(expected), truncatingFormat(actual)
+
+	return truncatingFormat(expected, true), truncatingFormat(actual, true)
 }
 
 // truncatingFormat formats the data and truncates it if it's too long.
 //
 // This helps keep formatted error messages lines from exceeding the
 // bufio.MaxScanTokenSize max line length that the go testing framework imposes.
-func truncatingFormat(data interface{}) string {
-	value := fmt.Sprintf("%#v", data)
+//
+// If the `printValueOfBasicTypes` flag is set to true, the underlying value of
+// pointers of basic types (see `basicTypes` for list of basic types) will be
+// returned in addition to the pointer address. For non-basic types, only the pointer
+// address will be returned.
+// If the `printValueOfBasicTypes` flag is set to false, only the pointer address
+// will be returned, regardless of whether `data` is a basic type or not.
+func truncatingFormat(data interface{}, printValueOfBasicTypes bool) string {
+	var value string
+	if printValueOfBasicTypes && reflect.ValueOf(data).Kind() == reflect.Ptr {
+		v := reflect.ValueOf(data).Elem()
+		for _, t := range basicTypes {
+			if v.Kind() == t {
+				value = fmt.Sprintf("%#v %v", data, v.Interface())
+				break
+			}
+		}
```

Comment:
I'm not against your `basicTypes` but this could be written as:
```suggestion
	if isBasicType(v.Kind()) {
		value = fmt.Sprintf("%#v %v", data, v.Interface())
	}
```

With:
```go
func isBasicType(k reflect.Kind) bool {
	return k == reflect.String || (k >= reflect.Bool && k <= reflect.Complex128)
}
```

Rather than iterating.